store/src/java/com/zimbra/cs/service/admin/GetDomainInfo.java

```diff
@@ -129,8 +134,13 @@ private void addAttrElementIfNotNull(Entry entry, Element element, String key) t
                 element.addNonUniqueElement(AdminConstants.E_A).addAttribute(AdminConstants.A_N, key).setText(value);
             }
         } else {
+            // Checking whether Modern UI is installed
+            File modernUIFile = new File(modernUIFolderPath);
             String value = entry.getAttr(key, null);
-            if (value != null) {
+            if (key.equals(ZAttrProvisioning.A_zimbraModernWebClientDisabled) && !(modernUIFile.exists() && modernUIFile.isDirectory())) {
```

Comment:
shouldn't this be OR condition when zimbraModernWebClientDisabled is TRUE or modern UI is not installed in either of cases response should be TRUE. Also if zimbraModernWebClientDisabled is TRUE we won't need to check it right as that's explicitly disable even if the Modern UI exists